crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -337,7 +340,10 @@ pub fn op_to_record(op: Op, omitted_entry: Option<Entry>) -> Record {
     }
 }
 
-async fn dhtop_to_op(op: DhtOp, cascade: &impl Cascade) -> AppValidationOutcome<Op> {
+async fn dhtop_to_op<C>(op: DhtOp, cascade: Arc<C>) -> AppValidationOutcome<Op>
```

Comment:
Is there a reason you made this generic? It think it's still possible to use `Arc<impl Cascade>`, right? If not, even `Arc<dyn Cascade>` might be preferable for the ergonomic benefit. I hope to use traits in a lot more places in the future, and it's nice to not have generics infect the whole codebase unless absolutely necessary, since they're syntactically awkward and a bit demoralizing IMO

Reply:
I had some trouble refactoring this, ended up with a more complicated generic argument, then figured out how to tidy it up. It was a whole saga :) Yes this can go back to just being an impl Trait argument. I'll go through and find these